From 653e191d55df893d3080dbc53f45bdbb7ba5acbb Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 23 May 2017 20:06:40 +0800
Subject: [PATCH 1805/5242] MLK-14967-2 ARM64: dts: freescale: imx8qxp:
 reserve DDR for M4

commit  c2e9418d3b51a919cb610c52e2f4d708a9391a57 from
https://source.codeaurora.org/external/imx/linux-imx.git

Reserve first 4M memory for below usage:

0x80000000 - 0x800fffff: ARM Trusted Firmware;
0x80100000 - 0x803fffff: M4/RPMSG;

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index 022ac5c..7eca843 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -15,7 +15,9 @@
 /dts-v1/;
 
 /* First 128KB is for PSCI ATF. */
-/memreserve/ 0x80000000 0x00020000;
+/* Last 3M is for M4/RPMSG */
+/memreserve/ 0x80000000 0x00400000;
+
 #include "fsl-imx8qxp.dtsi"
 
 / {
-- 
1.7.9.5

